react-app/src/app/components/Topbar/index.js

```diff
@@ -15,6 +15,7 @@ class Topbar extends Component {
           exact
           activeClassName={style.topbarActive}
           className={style.topbarLink}
+          onClick={link.isLogout && this.props.logout}
```

Comment:
I would define a `onClick` prop that receives a handler. Doing this will work for now, but it wont scale since you would need to add a condition for each icon type that needs to execute a function. 
Also, you can extract this method to a new component, in order to keep this component as clean as possible and split responsabilities